### ISSUE TITLE
caching and geo_format

### ISSUE THREAD

**Issue by mountainMath** [mountainMath/cancensus#79]
There seems to be an issue with caching data and geo_format. The current implementation stores files using the geo_format specified in the first call. If a subsequent call requests the same geographic data in a different geo_format the call fails.

**Comment by mountainMath** [comment]
One way around this is to go back to storing the geographic data as csv.

**Comment by mountainMath** [comment]
Not that urgent. Any reason why we won't just store the geojson? Performance? I guess this won't trigger any more dependencies, if people request the data in both formats they will have all the required libraries to do the conversion.

**Comment by atheriel** [comment]
I'm not really sure whether the performance improvements are significant. The space saved by serializing to R's data format over CSV is quite significant, but I'm not sure the same can be said of the GeoJSON files.

**Comment by mountainMath** [comment]
Another potential issue around our new caching regimen is that the cache gets wiped every time we re-install the package. Not sure if that's desirable, people's API points will take a hit if they re-install while working on something and they use the default caching scheme.

**Comment by atheriel** [comment]
Perhaps the way to do this is notify the user about the local cache option. E.g. so we'd have

```r
library(cancensus)
```
    ## Using internal package cache, which will be wiped on update.
    ## Consider using options(cancensus.cache_path = 'XXX') instead.


We could also add some logic to check both the internal cache and the locally-set one for the relevant data.

Of course, this doesn't get around the issue that users who have more than one local cache will *also* hammer their API quota unnecessarily.

**Comment by mountainMath** [comment]
I like the message on load to coax people into specifying a cache directory.

For the fallback setting, what's the lesser of the two evils? Project by project local cache or built-by-built global cache?

**Comment by mountainMath** [comment]
Ok, I might have just goofed. I merged the sf <-> sp conversion, but than ran into problems. So I reverted the merge, might have to open the pull request again.

Issues seemed to revolve around stored `sf` objects not properly serializing/de-serializing. Why don't we go back to storing geojson for now? Less efficient in terms of computation and storage, but more robust.

If there are no objections I will work in a pull request this afternoon.
